### PR TITLE
 feat: Implement Enterprise-Grade Auth, Logging, and API Gateway

### PR DESCRIPTION
**Pull request by kushal45**
This pull request introduces a major overhaul of the authentication, logging, and API gateway systems, transforming the project into a robust, enterprise-grade platform. It includes a complete authentication service, a comprehensive logging infrastructure, a dynamic API gateway, and extensive documentation and testing.

### 1.  **Enterprise Logging Infrastructure**

*   **Structured Logging:** All log entries are now in a consistent JSON format with detailed metadata, including correlation IDs, service names, and request context.
*   **Correlation IDs:** Enables end-to-end request tracing across all microservices.
*   **Multiple Transports:** Supports console, file, and daily rotating file transports, with configurable log levels and retention policies.
*   **Security:** Includes automatic redaction of sensitive data (passwords, tokens, etc.) in logs.
*   **Performance Monitoring:** Provides utilities for timing operations and tracking performance metrics.

### 2.  **Dynamic API Gateway with Service Discovery**

*   **Dynamic Route Configuration:** The API Gateway now loads its routing configuration from an external JSON file (`config/route-mappings.json`), eliminating hardcoded routes.
*   **Environment Variable Interpolation:** Supports dynamic configuration using environment variables with default fallbacks (e.g., `${VAR_NAME:default_value}`).
*   **Schema Validation:** A JSON schema (`config/route-mapping.schema.json`) ensures the integrity of the route mapping configuration.
*   **Intelligent Routing:** The gateway uses a priority-based system to match incoming requests to the appropriate microservices, with support for regex, glob, and exact-match patterns.
*   **Comprehensive Error Handling:** Provides standardized error responses for common gateway issues (404, 502, 503, 504).

### 3.  **Full-Featured Authentication Service**

*   **JWT-Based Authentication:** A complete implementation of JWT-based authentication, including login, logout, token refresh, and token validation.
*   **User Registration:** A secure user registration flow for both system admins and tenant admins.
*   **Password Management:** Secure password hashing using `bcrypt` and password strength validation.
*   **Security Hardening:** Includes rate limiting on login attempts to prevent brute-force attacks.
*   **Database Integration:** All authentication data is stored in the database, with new tables for `auth_credentials` and `auth_tokens`.

### 4.  **Database Schema and Migrations**

*   **New Auth Tables:** Introduced `auth_credentials` and `auth_tokens` tables to manage authentication data securely.
*   **User Table Enhancements:** The `users` table has been updated to support different user types (`system_admin`, `tenant_admin`, etc.) and to remove sensitive data like password hashes.
*   **TypeORM Migrations:** Includes two new database migrations to apply the schema changes.

### 5.  **Extensive Documentation**

*   **API Gateway Guide:** A comprehensive guide to the new API Gateway, including its architecture, endpoints, and configuration.
*   **Enterprise Logging Guide:** Detailed documentation for the new logging system, covering its features, configuration, and best practices.
*   **Dynamic Route Configuration:** A guide to the new dynamic route mapping system.
*   **Debugging Guide:** A new guide for debugging the applications and tests using the provided VS Code launch configurations.

### 6.  **Comprehensive Testing**

*   **Admin Registration E2E Tests:** A new suite of end-to-end tests for the admin registration flow, covering various scenarios and edge cases.
*   **Unit and Integration Tests:** Numerous new unit and integration tests have been added for the new services and components.

### 7.  **Improved Developer Experience**

*   **VS Code Debugging:** A set of pre-configured VS Code launch configurations for debugging all services and tests.
*   **Environment Setup Script:** A new `setup-env.sh` script to automatically create `.env` files from the examples.
*   **Linting and Formatting:** Improved ESLint and Prettier configurations to ensure code quality and consistency.

## How to Test:

1.  **Set up the environment:**
    ```bash
    ./setup-env.sh
    ```
2.  **Install dependencies:**
    ```bash
    npm install
    ```
3.  **Run the database migrations:**
    ```bash
    npm run migration:run
    ```
4.  **Start the services:**
    ```bash
    npm run start:dev
    ```
5.  **Run the tests:**
    ```bash
    npm test
    ```
6.  **Run the E2E tests:**
    ```bash
    npm run test:e2e
    ```

This pull request represents a significant step forward in the project's maturity, providing a solid foundation for future development and enterprise adoption.

### CODE REVIEW

File: apps/auth-service/test/helpers/test-data.factory.ts [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+   * Clean up test data
+   */
+  async cleanupTestData(): Promise<void> {
+    // Clean up any test-specific data
+    // This could be expanded to remove specific test records
+    const userRepository = this.dataSource.getRepository('User');
+    await userRepository.delete({ email: { $like: '%@rbactest.com' } as unknown as string });
+
+    const tenantRepository = this.dataSource.getRepository('Tenant');
+    await tenantRepository.delete({ slug: { $like: 'test-tenant-%' } as unknown as string });
```
[the remark this answers is not in the document]
Comment (kushal45) [reply apps/auth-service/test/helpers/test-data.factory.ts:329]: @copilot  couldd you please be more elaborate and provide the correct code here ?
